transformer_engine/pytorch/fp8.py

```diff
@@ -293,8 +298,12 @@ def fp8_autocast(
             assert fp8_available, reason_for_no_fp8
         yield
     finally:
-        _FP8_ENABLED,_FP8_CALIBRATION, _FP8_RECIPE, _FP8_DISTRIBUTED_GROUP = fp8_state
-        _IS_FIRST_FP8_MODULE = False
+        (_FP8_ENABLED,
+         _FP8_CALIBRATION,
+         _FP8_RECIPE,
+         _FP8_DISTRIBUTED_GROUP,
+         _IS_FIRST_FP8_MODULE) = fp8_state
```

Comment:
Could we run into a bug if the user does something like:
```python
with fp8_autocast():
    with fp8_autocast():
        module1()  # is_first_fp8_module() == True
    with fp8_autocast():
        module2()  # is_first_fp8_module() == True
```
I think the existing implementation also has a bad edge case:
```python
with fp8_autocast():
    with fp8_autocast():
        pass
    with fp8_autocast():
        module1(x)  # is_first_fp8_module() == False
```
I think the right answer is to remove `_IS_FIRST_FP8_MODULE` entirely from the `finally` clause, but I'm not confident if there are other interactions I'm missing.